kano_settings/system/wallpaper.py

```diff
@@ -2,36 +2,79 @@
 
 # wallpaper.py
 #
-# Copyright (C) 2014 Kano Computing Ltd.
-# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
+# Copyright (C) 2014-2016 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
 #
 # Backend wallpaper functions
 #
 
 
 from kano.logging import logger
+from kano.utils import get_user_unsudoed
 import os
 
-KDESKRC_HOME = "/home/{user}/.kdeskrc"
+KDESKRC_SYSTEM = "/usr/share/kano-desktop/kdesk/.kdeskrc"
+KDESKRC_HOME_TEMPLATE = "/home/{user}/.kdeskrc"
+
+PARAM_NAME_TEMPLATE = "Background.File-{size}:"
+SIZE_SUFFIX_MAP = {
+    "medium": "-1024.png",
+    "4-3": "-4-3.png",
+    "16-9": "-16-9.png"
+}
+
+
+def _get_wallpaper_from_kdeskrc(kdeskrc_path):
+    """
+        Parses a .kdeskrc file and looks for wallpaper configuration keys.
+
+        :param kdeskrc_path: Location of the file to be parsed.
+        :type kdeskrc_path: str
+
+        :return: A map of wallpaper types to file paths.
+        :rtype: dict
+    """
+
+    wallpapers = {}
+    if os.path.isfile(kdeskrc_path):
+        with open(kdeskrc_path, 'r') as kdesk_conf:
+            for kdesk_conf_line in kdesk_conf:
+                for key, suffix in SIZE_SUFFIX_MAP.iteritems():
+                    token = "{}:".format(key)
+                    if token in kdesk_conf_line:
+                        path = kdesk_conf_line.split(token)[-1].strip()
+                        wallpapers[key] = path
```

Comment:
This looks a bit Zsolty but it does seem to do the job and I can't find an easy solution to the problem (other than creating a function for something like `parse_kdesk_conf_line(conf_line)`)